### PR TITLE
Add verification config types from RFC

### PR DESCRIPTION
**Pull request by viccuad**
Currently, the `Subject` enum types use both `#[serde(deny_unknown_fields)]` and `#[serde(flatten)] `, which is not supported, and is supposed to fail on deserialization. Strangely, it works here, as the tests pass.

Also tested by consuming `verify::read_verification_file()` from a simple crate, instead of consuming this branch from policy-server and policy-evaluator.

Part of https://github.com/kubewarden/policy-server/issues/142.

### CODE REVIEW

**Comment by viccuad** [comment]
Force pushed to rebase main changes.

**Comment by viccuad** [comment]
rebased main again.

File: src/verify/config.rs [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+    pub signatures: Vec<Signature>,
+}
+
+fn default_minimum_matches() -> String {
+    "1".to_string()
+}
+
+#[derive(Deserialize, Debug, Clone, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+#[serde(tag = "kind")]
```
Comment (ereslibre) [inline src/verify/config.rs:30]:
```suggestion
#[serde(rename_all = "camelCase", tag = "kind")]
```

**Review by ereslibre — APPROVED** [review]
LGTM. Minor comment, feel free to ignore if you want.

File: src/verify/config.rs (hunk 1 above)
Comment (viccuad) [reply src/verify/config.rs]: Thanks! done in https://github.com/kubewarden/policy-fetcher/pull/42/commits/3e745f0639f1adcd811afc5b5cc15bef88e6f040

File: src/verify/config.rs [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+        // `https://github.com/kubewarden` matching
+        // `https://github.com/kubewarden-malicious/`
+        url.set_path(format!("{}{}", url.path(), '/').as_str());
+    }
+    Ok(url)
+}
+
+#[derive(Deserialize, Debug, Clone, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+#[serde(deny_unknown_fields)]
```
[the remark this answers is not in the document]
Comment (viccuad) [reply src/verify/config.rs]: Thanks! done in https://github.com/kubewarden/policy-fetcher/pull/42/commits/3e745f0639f1adcd811afc5b5cc15bef88e6f040

**Comment by viccuad** [comment]
Please re-review, I found out I misread the RFC, and I wasn't covering the case where `any_of` is not listed.

**Review by flavio — COMMENTED** [review]
Sorry, I missed this PR. I left some comments about it inside of https://github.com/kubewarden/policy-fetcher/pull/50 (because #50 is built on top of this one)

**Comment by flavio** [comment]
Superseded by https://github.com/kubewarden/policy-fetcher/pull/54

**Pull request closed without merging**